Assets/MapzenMap.cs

```diff
@@ -35,8 +37,11 @@ void Start()
 
                 Debug.Log("URL request " + url);
 
-                HTTPRequestCallback callback = (string error, string response, TileAddress address) =>
+                HTTPRequestCallback callback = (string error, string response, IntPtr userData) =>
                 {
+                    GCHandle addressHandle = GCHandle.FromIntPtr(userData);
```

Comment:
I don't think that we need to create a `GCHandle` in between to store the tile address and manage memory manually.

For a more generic interface, what about something on those lines?

```c#
class HTTPRequest {
    delegate void HTTPRequestCallback(object userData, Response httpResponse);

    public HTTPRequest(string url, HTTPRequestCallback callback);
}
```

Where `Reponse` object has the following simple data structure:

```c#	public struct Response
class Response {
    public string Error;
    public byte[] Data;
}
```